workers/caching/mpcache.ts

```diff
@@ -0,0 +1,116 @@
+/// <reference types="@cloudflare/workers-types" />
+
+export default {
+  async fetch(request, env, ctx) {
+    const cacheKeyGeneration = env.CACHE_KEY_GENERATION || "v1"
+    const defaultMaxAge = env.DEFAULT_MAX_AGE || 30
+
+    // Returns a SHA-256 digest of provided string.
+    async function sha256(message) {
+      // encode as UTF-8
+      const msgBuffer = await new TextEncoder().encode(message)
+
+      // hash the message
+      const hashBuffer = await crypto.subtle.digest("SHA-256", msgBuffer)
+
+      // convert bytes to hex string
+      return [...new Uint8Array(hashBuffer)]
+        .map((b) => b.toString(16).padStart(2, "0"))
+        .join("")
+    }
+
+    // Construct a GET request with digest appended to path, to serve as cache key.
+    async function getCacheKey(request) {
```

Comment:
can't we move the smaller functions outside? it'd make it better to read the `default` block